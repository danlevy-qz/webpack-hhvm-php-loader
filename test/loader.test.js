import {
  compile,
  execute,
  getCompiler,
  normalizeErrors,
  readAsset,
} from './helpers';

describe('loader', () => {
  it('should work', async () => {
    const compiler = getCompiler('simple.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should fail on bad paths', async () => {
    const compiler = getCompiler('bad-script.js');

    // expect.assertions(1);

    return compile(compiler)
      .then((results) => {
        expect(results).toBeNull();
      })
      .catch((error) => {
        expect(error.runtimeTemplate).toMatchSnapshot('bad-path-failure');
      });
  });

  it('should work with JSON', async () => {
    const compiler = getCompiler('json.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with inline loader', async () => {
    const compiler = getCompiler('inline.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "ModuleConcatenationPlugin" plugin', async () => {
    const compiler = getCompiler(
      'simple.js',
      {},
      {
        mode: 'production',
        optimization: {
          minimize: false,
        },
      }
    );
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');

    if (stats.compilation.modules.size) {
      expect(stats.compilation.modules.size).toBe(2);
    } else {
      expect(stats.compilation.modules.length).toBe(1);
    }
  });
});
