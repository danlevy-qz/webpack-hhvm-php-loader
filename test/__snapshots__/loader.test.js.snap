// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should fail on bad paths: bad-path-failure 1`] = `undefined`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: errors 1`] = `Array []`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: result 1`] = `"Hello World"`;

exports[`loader should work with "ModuleConcatenationPlugin" plugin: warnings 1`] = `Array []`;

exports[`loader should work with JSON: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with JSON: errors 1`] = `Array []`;

exports[`loader should work with JSON: result 1`] = `
Object {
  "date_created": "2020-12-01",
  "first_name": "Dan",
  "username": "justsml",
}
`;

exports[`loader should work with JSON: warnings 1`] = `Array []`;

exports[`loader should work with inline loader: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work with inline loader: errors 1`] = `Array []`;

exports[`loader should work with inline loader: result 1`] = `"Hello World"`;

exports[`loader should work with inline loader: warnings 1`] = `Array []`;

exports[`loader should work: assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`loader should work: errors 1`] = `Array []`;

exports[`loader should work: result 1`] = `"Hello World"`;

exports[`loader should work: warnings 1`] = `Array []`;
